nominal/_utils.py

```diff
@@ -246,3 +139,47 @@ def update_dataclass(self: T, other: T, fields: Iterable[str]) -> None:
     """
     for field in fields:
         self.__dict__[field] = getattr(other, field)
+
+
+class FileType(NamedTuple):
```

Comment:
why `NamedTuple` vs frozen dataclass? (just curious, only loosely familiar with them)

Reply:
Ah good point. Generally I use them for the opposite reason I did here - to give structure to an API that you can otherwise pass a tuple, e.g. pass me a Point(x, y) so that I can shuttle it to an underlying API that only knows about tuples. But that's not the case here.